aries/aries_vcx_core/src/anoncreds/anoncreds.rs

```diff
@@ -0,0 +1,1799 @@
+    async fn verifier_verify_proof(
+        &self,
+        proof_request_json: &str,
+        proof_json: &str,
+        schemas_json: &str,
+        credential_defs_json: &str,
+        rev_reg_defs_json: &str,
+        rev_regs_json: &str,
+    ) -> VcxCoreResult<bool> {
+        let presentation: Presentation = serde_json::from_str(proof_json)?;
+        let pres_req: PresentationRequest = serde_json::from_str(proof_request_json)?;
+
+        let mut schemas_val: HashMap<SchemaId, Value> = serde_json::from_str(schemas_json)?;
+        let mut schemas: HashMap<SchemaId, Schema> = HashMap::new();
+        for (schema_id, schema_json) in schemas_val.iter_mut() {
+            if let Some(v) = schema_json.as_object_mut() {
+                v.insert(
+                    "issuerId".to_owned(),
+                    schema_parts(&schema_id.to_string()).unwrap().1.did().into(),
+                );
+            }
+            let schema = serde_json::from_value(schema_json.clone())?;
+            schemas.insert(schema_id.clone(), schema);
+        }
+
+        let mut cred_defs_val: HashMap<CredentialDefinitionId, Value> =
+            serde_json::from_str(credential_defs_json)?;
+        let mut cred_defs: HashMap<CredentialDefinitionId, CredentialDefinition> = HashMap::new();
+        for (cred_def_id, cred_def_json) in cred_defs_val.iter_mut() {
+            if let Some(v) = cred_def_json.as_object_mut() {
+                v.insert(
+                    "issuerId".to_owned(),
+                    cred_def_parts(&cred_def_id.to_string())
+                        .unwrap()
+                        .1
+                        .did()
+                        .into(),
+                );
+            }
+            let cred_def = serde_json::from_value(cred_def_json.clone())?;
+            cred_defs.insert(cred_def_id.clone(), cred_def);
+        }
+
+        let rev_reg_defs_val: Option<HashMap<RevocationRegistryDefinitionId, Value>> =
+            serde_json::from_str(rev_reg_defs_json)?;
+        let rev_reg_defs: Option<
+            HashMap<RevocationRegistryDefinitionId, RevocationRegistryDefinition>,
+        > = if let Some(mut rev_defs) = rev_reg_defs_val.clone() {
+            let mut map = HashMap::new();
+            for (rev_reg_def_id, rev_reg_def_json) in rev_defs.iter_mut() {
+                let v = rev_reg_def_json.as_object_mut().unwrap();
+                v.insert(
+                    "issuerId".to_owned(),
+                    rev_reg_def_id.to_string().split(':').next().into(),
+                );
+                v.insert("revocDefType".to_owned(), CL_ACCUM.to_owned().into());
+
+                map.insert(
+                    rev_reg_def_id.clone(),
+                    serde_json::from_value(rev_reg_def_json.clone())?,
+                );
+            }
+            Some(map)
+        } else {
+            None
+        };
+
+        // WTF, anoncreds-rs expects the whole status list just to use the accumulator :|
```

Comment:
As discussed, it may be fine to keep the API as thin as possible type-wise. It needs to be carefully considered what changes to anoncreds-rs are necessary. Not my priority right now.

Reply:
As I said, at this point, I only know some changes will be necessary (which is obviated by the fact we are using custom fork - hence no need to track IMO), but I don't know which ones. This question will get deserved attention later.